apps/terminal/serializers/endpoint.py

```diff
@@ -59,7 +59,7 @@ def get_extra_kwargs(self):
         return extra_kwargs
 
     def validate_is_active(self, value):
-        if str(self.instance.id) == Endpoint.default_id:
+        if self.instance and str(self.instance.id) == Endpoint.default_id:
             # 默认端点不能禁用
             return True
         else:
```

Comment:
There are no major discrepancies noted between this code and other versions provided. It seems to be functioning properly based on the knowledge cut off at 2021-09-01. However, it would still perform better through updates to support recent Python and API version compatibility changes. For more modern usage scenarios or when dealing with complex functionalities that may evolve in newer releases of APIs (such as endpoints), consider refactoring into a separate library layer where developers can import and use classes like `Endpoint` without worrying about backward-compatibility.

For an update suggestion towards the latest standards:

**Add explicit type annotations**

```
class Meta:
    """Metadata for Endpoint objects."""
    
```